var/spack/repos/builtin/packages/lzo/package.py

```diff
@@ -37,13 +37,14 @@ class Lzo(Package):
     version('2.06', '95380bd4081f85ef08c5209f4107e9f8')
     version('2.05', 'c67cda5fa191bab761c7cb06fe091e36')
 
-    def install(self, spec, prefix):
-        configure_args = [
-            '--prefix={0}'.format(prefix),
+    def configure_args(self):
+        return [
             '--disable-dependency-tracking',
             '--enable-shared'
         ]
-        configure(*configure_args)
-        make()
+
+    @AutotoolsPackage.sanity_check('build')
+    @AutotoolsPackage.on_package_attributes(run_tests=True)
+    def check(self):
```

Comment:
@davydden @adamjstewart @tgamblin Here's an example how we can deal with standard checks and enable them based on the value of an attribute. A few things : 
- the decorator here is entirely optional, and is there just to show that we can leverage a similar syntax
- if we go this way we may add some smartness to standard ways of checking (like `make check` and `make test`) and push them to the base class